File: src/main/scala-graphql/io/github/graphql/model/OrganizationIdentityProviderResponseProjection.scala
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection

/**
 * Response projection for OrganizationIdentityProvider
 */
@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:00+0800"
)
class OrganizationIdentityProviderResponseProjection extends GraphQLResponseProjection {

    override def all$(): OrganizationIdentityProviderResponseProjection = all$(3)

    override def all$(maxDepth: Int): OrganizationIdentityProviderResponseProjection = {
        this.digestMethod()
        if (projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.ExternalIdentityConnectionResponseProjection.externalIdentities", 0) <= maxDepth) {
            projectionDepthOnFields.put("OrganizationIdentityProviderResponseProjection.ExternalIdentityConnectionResponseProjection.externalIdentities", projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.ExternalIdentityConnectionResponseProjection.externalIdentities", 0) + 1)
            this.externalIdentities(new ExternalIdentityConnectionResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.ExternalIdentityConnectionResponseProjection.externalIdentities", 0)))
        }
        this.id()
        this.idpCertificate()
        this.issuer()
        if (projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.OrganizationResponseProjection.organization", 0) <= maxDepth) {
            projectionDepthOnFields.put("OrganizationIdentityProviderResponseProjection.OrganizationResponseProjection.organization", projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.OrganizationResponseProjection.organization", 0) + 1)
            this.organization(new OrganizationResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.OrganizationResponseProjection.organization", 0)))
        }
        this.signatureMethod()
        this.ssoUrl()
        this.typename()
        this
    }

    def digestMethod(): OrganizationIdentityProviderResponseProjection = {
        digestMethod(null.asInstanceOf[String])
    }

    def digestMethod(alias: String): OrganizationIdentityProviderResponseProjection = {
        fields.add(new GraphQLResponseField("digestMethod").alias(alias))
        this
    }

    def externalIdentities(subProjection: ExternalIdentityConnectionResponseProjection): OrganizationIdentityProviderResponseProjection = {
        externalIdentities(null.asInstanceOf[String], subProjection)
    }

    def externalIdentities(alias: String, subProjection: ExternalIdentityConnectionResponseProjection): OrganizationIdentityProviderResponseProjection = {
        fields.add(new GraphQLResponseField("externalIdentities").alias(alias).projection(subProjection))
        this
    }

    def externalIdentities(input: OrganizationIdentityProviderExternalIdentitiesParametrizedInput,subProjection: ExternalIdentityConnectionResponseProjection): OrganizationIdentityProviderResponseProjection = {
        externalIdentities(null.asInstanceOf[String], input, subProjection)
    }

    def externalIdentities(alias: String, input: OrganizationIdentityProviderExternalIdentitiesParametrizedInput , subProjection: ExternalIdentityConnectionResponseProjection): OrganizationIdentityProviderResponseProjection = {
        fields.add(new GraphQLResponseField("externalIdentities").alias(alias).parameters(input).projection(subProjection))
        this
    }

    def id(): OrganizationIdentityProviderResponseProjection = {
        id(null.asInstanceOf[String])
    }

    def id(alias: String): OrganizationIdentityProviderResponseProjection = {
        fields.add(new GraphQLResponseField("id").alias(alias))
        this
    }

    def idpCertificate(): OrganizationIdentityProviderResponseProjection = {
        idpCertificate(null.asInstanceOf[String])
    }

    def idpCertificate(alias: String): OrganizationIdentityProviderResponseProjection = {
        fields.add(new GraphQLResponseField("idpCertificate").alias(alias))
        this
    }

    def issuer(): OrganizationIdentityProviderResponseProjection = {
        issuer(null.asInstanceOf[String])
    }

    def issuer(alias: String): OrganizationIdentityProviderResponseProjection = {
        fields.add(new GraphQLResponseField("issuer").alias(alias))
        this
    }

    def organization(subProjection: OrganizationResponseProjection): OrganizationIdentityProviderResponseProjection = {
        organization(null.asInstanceOf[String], subProjection)
    }

    def organization(alias: String, subProjection: OrganizationResponseProjection): OrganizationIdentityProviderResponseProjection = {
        fields.add(new GraphQLResponseField("organization").alias(alias).projection(subProjection))
        this
    }

    def signatureMethod(): OrganizationIdentityProviderResponseProjection = {
        signatureMethod(null.asInstanceOf[String])
    }

    def signatureMethod(alias: String): OrganizationIdentityProviderResponseProjection = {
        fields.add(new GraphQLResponseField("signatureMethod").alias(alias))
        this
    }

    def ssoUrl(): OrganizationIdentityProviderResponseProjection = {
        ssoUrl(null.asInstanceOf[String])
    }

    def ssoUrl(alias: String): OrganizationIdentityProviderResponseProjection = {
        fields.add(new GraphQLResponseField("ssoUrl").alias(alias))
        this
    }

    def typename(): OrganizationIdentityProviderResponseProjection = {
        typename(null.asInstanceOf[String])
    }

    def typename(alias: String): OrganizationIdentityProviderResponseProjection = {
        fields.add(new GraphQLResponseField("__typename").alias(alias))
        this
    }


}

File: src/main/kotlin-graphql/io/github/graphql/model/OrganizationIdentityProviderResponseProjection.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection

/**
 * Response projection for OrganizationIdentityProvider
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-06T19:48:03+0800"
)
open class OrganizationIdentityProviderResponseProjection : GraphQLResponseProjection() {

    override fun `all$`(): OrganizationIdentityProviderResponseProjection = `all$`(3)

    override fun `all$`(maxDepth: Int): OrganizationIdentityProviderResponseProjection {
        this.digestMethod()
        if (projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.ExternalIdentityConnectionResponseProjection.externalIdentities", 0) <= maxDepth) {
            projectionDepthOnFields["OrganizationIdentityProviderResponseProjection.ExternalIdentityConnectionResponseProjection.externalIdentities"] = projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.ExternalIdentityConnectionResponseProjection.externalIdentities", 0) + 1
            this.externalIdentities(ExternalIdentityConnectionResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.ExternalIdentityConnectionResponseProjection.externalIdentities", 0)))
        }
        this.id()
        this.idpCertificate()
        this.issuer()
        if (projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.OrganizationResponseProjection.organization", 0) <= maxDepth) {
            projectionDepthOnFields["OrganizationIdentityProviderResponseProjection.OrganizationResponseProjection.organization"] = projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.OrganizationResponseProjection.organization", 0) + 1
            this.organization(OrganizationResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("OrganizationIdentityProviderResponseProjection.OrganizationResponseProjection.organization", 0)))
        }
        this.signatureMethod()
        this.ssoUrl()
        this.typename()
        return this
    }

    fun digestMethod(): OrganizationIdentityProviderResponseProjection = digestMethod(null)

    fun digestMethod(alias: String?): OrganizationIdentityProviderResponseProjection {
        fields.add(GraphQLResponseField("digestMethod").alias(alias))
        return this
    }

    fun externalIdentities(subProjection: ExternalIdentityConnectionResponseProjection): OrganizationIdentityProviderResponseProjection = externalIdentities(null, subProjection)

    fun externalIdentities(alias: String?, subProjection: ExternalIdentityConnectionResponseProjection): OrganizationIdentityProviderResponseProjection {
        fields.add(GraphQLResponseField("externalIdentities").alias(alias).projection(subProjection))
        return this
    }

    fun externalIdentities(input: OrganizationIdentityProviderExternalIdentitiesParametrizedInput, subProjection: ExternalIdentityConnectionResponseProjection): OrganizationIdentityProviderResponseProjection = externalIdentities(null, input, subProjection)

    fun externalIdentities(alias: String?, input: OrganizationIdentityProviderExternalIdentitiesParametrizedInput, subProjection: ExternalIdentityConnectionResponseProjection): OrganizationIdentityProviderResponseProjection {
        fields.add(GraphQLResponseField("externalIdentities").alias(alias).parameters(input).projection(subProjection))
        return this
    }

    fun id(): OrganizationIdentityProviderResponseProjection = id(null)

    fun id(alias: String?): OrganizationIdentityProviderResponseProjection {
        fields.add(GraphQLResponseField("id").alias(alias))
        return this
    }

    fun idpCertificate(): OrganizationIdentityProviderResponseProjection = idpCertificate(null)

    fun idpCertificate(alias: String?): OrganizationIdentityProviderResponseProjection {
        fields.add(GraphQLResponseField("idpCertificate").alias(alias))
        return this
    }

    fun issuer(): OrganizationIdentityProviderResponseProjection = issuer(null)

    fun issuer(alias: String?): OrganizationIdentityProviderResponseProjection {
        fields.add(GraphQLResponseField("issuer").alias(alias))
        return this
    }

    fun organization(subProjection: OrganizationResponseProjection): OrganizationIdentityProviderResponseProjection = organization(null, subProjection)

    fun organization(alias: String?, subProjection: OrganizationResponseProjection): OrganizationIdentityProviderResponseProjection {
        fields.add(GraphQLResponseField("organization").alias(alias).projection(subProjection))
        return this
    }

    fun signatureMethod(): OrganizationIdentityProviderResponseProjection = signatureMethod(null)

    fun signatureMethod(alias: String?): OrganizationIdentityProviderResponseProjection {
        fields.add(GraphQLResponseField("signatureMethod").alias(alias))
        return this
    }

    fun ssoUrl(): OrganizationIdentityProviderResponseProjection = ssoUrl(null)

    fun ssoUrl(alias: String?): OrganizationIdentityProviderResponseProjection {
        fields.add(GraphQLResponseField("ssoUrl").alias(alias))
        return this
    }

    fun typename(): OrganizationIdentityProviderResponseProjection = typename(null)

    fun typename(alias: String?): OrganizationIdentityProviderResponseProjection {
        fields.add(GraphQLResponseField("__typename").alias(alias))
        return this
    }


}
